mcs/class/corlib/System/TimeZoneInfo.cs

```diff
@@ -116,11 +116,11 @@ static TimeZoneInfo CreateLocal ()
 
 			try {
 				return FindSystemTimeZoneByFileName ("Local", "/etc/localtime");	
-			} catch {
+			} catch (TimeZoneNotFoundException) {
```

Comment:
To avoid introducing new issues this line should not change.
We don't want to ignore `Path.Combine (TimeZoneDirectory, "localtime")` when `/etc/localtime` is invalid.